### DISCUSSION TITLE
More H3 registers

### DISCUSSION THREAD

**Discussion by canton7** [TonyM1958/HA-FoxESS-Modbus#14] [General]
Hi @TonyM1958,

I've got some more H3 registers for you:

* 31042: BMS Connect State
* 31043: Meter Connect State
* 30100: Modbus protocol version
* 44000: Remote power control Enable
* 44001: Remote power control timeout (secs)
* 44002-44003: Remote Control-Active Power Command (watts, int32, +ve is charge, -ve is discharge)
* 44004-44005: Remote Control-Reactive Power Commad (VA, int32)

**Comment by TonyM1958** [comment]
I've updated the spreadsheet with the info for H3 and H1. 

For H1, there is both a Meter 1 Connection State and a Meter 2 Connection State - so I expect there is probably a Meter 2 Connection State register somewhere on H3 as well, although the consequentive addresses already seem to be used?

I assume you have already seen the remote control register entries and decodes in this doc: https://github.com/TonyM1958/HA-FoxESS-Modbus/files/11628403/Fox.-.AIO.H1.modbus.-.2023.pdf

**Reply by canton7** [reply]
Yep. I've got the H3 modbus spec as well.

**Reply by TonyM1958** [reply]
Nice. Can you share it?

**Reply by MartinVanLing** [reply]
I'm really looking for the H3 modbus spec, could you please share? THANK you!

**Reply by TonyM1958** [reply]
We don't actually have a complete spec (IMHO).

My best guess at the list of registers is contained in the modbus files - although not all of the registers listed have been verified (some are marked tbc).

Have a look through this file: https://github.com/TonyM1958/HA-FoxESS-Modbus/blob/main/custom_components/HA-FoxESS-Modbus/modbusH3_RS485.yaml

**Reply by canton7** [reply]
I've got the spec (well, for the AIO, which is close enough), but it was shared by someone in confidence, so I'm not going to post it publicly I'm afraid.

**Comment by MartinVanLing** [comment]
Thank you Tony, that is good info! Will work this to an XLS sheet for gateway config file.

**Reply by TonyM1958** [reply]
There is a spreadsheet here: https://github.com/TonyM1958/HA-FoxESS-Modbus/wiki/Modbus-Map.xls

I just haven't updated it recently so it could do with an overhaul. Most of what I've found out recently is in the various modbus files.

**Comment by MartinVanLing** [comment]
Awsome, thank you very much!!
Martin

**Comment by kinkate** [comment]
Hi there!
I've just finished my exploration for new registers. With some support from AI and "code search" in nathan's repo Ican add 6 registers for H3 -Converters. I tested these regs on an H3-12.0-E with success! I added it to my personal modbus.map.xlsx as Line 71 and on at the "Three Phase"-sheet (new RegNo's are red marked):
[modbus.map.xlsx](https://github.com/user-attachments/files/31090158/modbus.map.xlsx)

**Reply by TonyM1958** [reply]
Thanks - I've added these to the wiki and to the modbus YAML for H3  (v1.6.8).
